### PR TITLE
autogenview: store API parameters as router query for history

### LINKED ISSUE #451: Maintain page context in the URL
The page context for page, pagesize, keyword (maybe filter related things later on...) are not stored on the route/path/params and back/forward missing the context...

### PR DESCRIPTION
**Pull request by rhtyd**
This would store pagination, filter and search context in route query
which would allow for browser history navigation in list views.

Fixes #451

Signed-off-by: Rohit Yadav <rohit.yadav@shapeblue.com>

### CODE REVIEW

**Comment by rhtyd** [comment]
Please test and review @davidjumani

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
Converted to draft, as this may not be complete yet
With @davidjumani

**Comment by rhtyd** [comment]
@davidjumani no update/progress on this?

**Comment by davidjumani** [comment]
Have tested for VMs, Images and Networks. Testing the rest in progress

File: src/views/AutogenView.vue [hunk 1]
```diff
@@ -381,10 +382,19 @@ export default {
   watch: {
     '$route' (to, from) {
       if (to.fullPath !== from.fullPath && !to.fullPath.includes('action/')) {
-        this.searchQuery = ''
-        this.page = 1
+        if ('q' in to.query) {
```
Comment (davidjumani) [inline src/views/AutogenView.vue:385]: should pick up `query.keyword` instead ?

Comment (rhtyd) [reply src/views/AutogenView.vue:385]: I think not all views/APIs pass keyword as the query parameter (from search); for example global setting we use name

Comment (davidjumani) [reply src/views/AutogenView.vue:385]: Agreed but it creates two parameters keywork / name and q, with q having the higher priority, so changing only q and not the other can lead to inconsistencies. It by default picks up keywork / name when q is not present anyway

**Comment by davidjumani** [comment]
TODO once done, add the functionality to redirect to running / stopped vms via dasboard

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
Hold-on merging this, we may need more tests.

**Comment by rhtyd** [comment]
Ready for review and testing @davidjumani 
@blueorangutan package

**Comment by rhtyd** [comment]
Please review/test @davidjumani @Pearl1594 thanks

**Comment by rhtyd** [comment]
Did another round of test, please continue testing master instead @Pearl1594 @davidjumani

**Pull request merged**